source/cppfront.cpp

```diff
@@ -1771,6 +2034,7 @@ class cppfront
     //
     auto emit(
         type_id_node const& n,
+        bool                omit_typename = false,
```

Comment:
Still reading the code, but this name jumped out at me... above it's `should_emit` and `can_emit`, here it's `[should|can]_omit`. If it's the same thing with the sense inverted, and differing only by one letter, wouldn't it be better to be consistent and name it either `should_emit` or `can_emit` (including with the boolean sense inverted)?

Reply:
Just some initial thoughts... this is a big PR so I'll need more time to work through it. Thanks!

Reply:
I'll invert it.
I think I wanted to touch the least call sites possible.

Reply:
Instead, I went with `requires_typename_keyword` and `typename_keyword_is_optional`.

Reply:
Now there's
```C++
try_emit<type_id_node::qualified  >(n.id, !is_type_only_context && n.type_only_context);
```
where the former is Cpp1 and the latter Cpp2.